core/src/OAuth2/OAuth2Client.cs

```diff
@@ -166,7 +167,26 @@ public static void CreateErrorResponse(HttpResponse response, RequestContext req
                     ex);
             }
 
-            requestContext.Logger.ErrorPii(serviceEx);
+            // For device code flow, AuthorizationPending can occur a lot while waiting
+            // for the user to auth via browser and this causes a lot of error noise in the logs.
+            // So suppress this particular case to an Info so we still see the data but don't 
+            // log it as an error since it's expected behavior while waiting for the user.
+            bool shouldLogError = true;
+            if (serviceEx is CoreException coreException)
+            {
+                if (string.Compare(coreException.ErrorCode, OAuth2Error.AuthorizationPending,
```

Comment:
The exceptionCode seems to be local to this method, so not sure we really need to add all this logic (i.e. exposing a core exception is problematic for the reasons JM and Henrik mentioned, otherwise I agree that is the right solution for exposing the error code). 

Alternatively, we could add another method in the factory, something like ExceptionFactory.Instance.GetErrorCode where we would let each factory implementation pull out the error code.